impl/events.go

```diff
@@ -220,59 +162,78 @@ func (m *manager) OnResponseReceived(chid datatransfer.ChannelID, response datat
 				return err
 			}
 		}
+
+		// was the validateion attempt successful?
 		if !response.Accepted() {
+			// if not, error and fail
 			log.Infof("channel %s: received rejected response, erroring out channel", chid)
 			return m.channels.Error(chid, datatransfer.ErrRejected)
 		}
-		if response.IsNew() {
-			log.Infof("channel %s: received new response, accepting channel", chid)
-			err := m.channels.Accept(chid)
-			if err != nil {
-				return err
-			}
+	}
+
+	// was this the first response to our initial request
+	if response.IsNew() {
```

Comment:
this is interesting, both IsNew and IsRestart jumping outside of the previously containing `if`, was it wrong before not to check this in all but the IsComplete case?

Reply:
honestly they're just unneccesarily nested. IsNew = true & IsRestart = tree imply IsValidationResult == true, as does IsComplete = true.

So basically, the IsValidationResult handles the "!Accepted" condition and then the other things are determining behavior on an accepted response.